friends/src/components/FriendForm.js

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+
+
+export default class FriendForm extends Component {
+
+  constructor(props){
+    super(props);
+    this.state = {
+      name: '',
+      age: '',
+      email:''
+    }
+  }
+
+  
+
+  componentDidMount(prevProps, previousState){
+    let editFriend= this.props.editFriend;
+    console.log(editFriend);
+    if(editFriend.name){
+      this.setState({
+        name: editFriend.name,
+        age: editFriend.age,
+        email: editFriend.email
+      })
+    }
+  }
+  
+
+  handleChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  }
+
+  
+
+
+  render() {
+    return (
+      <div className="form">
+        <form onSubmit={e =>{
+          e.preventDefault();
+         
+          if(!this.props.editFriend.name){
+            // console.log('didnot')
+            const newFriendObj = {
+              name: this.state.name,
+              age: Number(this.state.age),
+              email: this.state.email,
```

Comment:
You could make this easier with object destructuring:
```javascript
const { name, age, email } = this.state;
// and then:
this.props.addToList({ name, age, email });
```
you could even technically do it this way since your state's just an object, but it's probably better to be more clear about it and also not pass the object itself:
```javascript
this.props.addToList(this.state);
```